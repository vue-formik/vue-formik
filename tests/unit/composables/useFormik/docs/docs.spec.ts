import { describe, expect, test } from "vitest";
import { useFormik } from "@/index";
import { InitialValues, ValidationSchemaYup, ValidationSchema } from "./fixture";

describe("docs", () => {
  test("yup", () => {
    const { values, errors, touched } = useFormik({
      initialValues: InitialValues,
      yupSchema: ValidationSchemaYup,
    });

    expect(values).toEqual(InitialValues);
    expect(errors).toMatchSnapshot();
    expect(touched).toMatchSnapshot();
  });
  test("custom validator", () => {
    const { values, errors, touched } = useFormik({
      initialValues: InitialValues,
      validationSchema: ValidationSchema,
    });

    expect(values).toEqual(InitialValues);
    expect(errors).toMatchSnapshot();
    expect(touched).toMatchSnapshot();
  });
});
