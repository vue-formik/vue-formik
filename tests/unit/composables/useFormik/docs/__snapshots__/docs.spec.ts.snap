// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`docs > custom validator 1`] = `
{
  "addresses": [
    "Address is required",
  ],
  "contacts": [
    {
      "code": "Code is required",
      "number": "Number is required",
    },
  ],
  "email": "Email is required",
  "name": "Name is required",
}
`;

exports[`docs > custom validator 2`] = `{}`;

exports[`docs > yup 1`] = `
{
  "addresses[0]": "Address is required",
  "contacts[0].code": "Code is required",
  "contacts[0].number": "Number is required",
  "email": "Email is required",
  "name": "Name is required",
}
`;

exports[`docs > yup 2`] = `{}`;
